Source/Scenes/World.cs

```diff
@@ -408,11 +426,24 @@ public override void Update()
 		{
 			if (Controls.Pause.Pressed || Controls.Cancel.Pressed)
 			{
-				SetPaused(false);
+				if (inSettingsMenu)
+				{
+					inSettingsMenu = false;
+				}
+				else
+				{
+					SetPaused(false);
+				}
 				Audio.Play(Sfx.ui_unpause);
 			}
 			else
-				pauseMenu.Update();
+			{
+				if (inSettingsMenu)
+				{
+					settingsMenu.Update();
+				}
+				else pauseMenu.Update();
+			}
```

Comment:
While this works for us, I think having a proper sub-menu system which could also be upsreamed/added as part of Fuji would be good.
Tho I'd be fine doing this for now (just add a TODO comment)